### PR TITLE
Feature/issue 10 controller get users

### LINKED ISSUE #10: Issue 10 : Controller Get User / id
**Priority :**
1

**Duration :** 
1

**Description:**
Creation of routes to list user / id

**File added / modified:**

**Additional context**

Add any other context or screenshots here.

### PR DESCRIPTION
**Pull request by vanmarcke**
Issue [#10]

Creation of routes to list user / id

### CODE REVIEW

File: src/Controller/UsersController.php [hunk 1]
```diff
@@ -33,4 +33,20 @@ public function showUsers(Request $request): JsonResponse
 
         return $this->json($usersList, Response::HTTP_OK);
     }
+
+    #[Route('/api/users/{id}', methods: ['GET'], name: 'user_show')]
+    public function showUser($id): JsonResponse
```
Comment (antoinetrouve) [inline src/Controller/UsersController.php]: Il faut typer ton $id

Comment (antoinetrouve) [reply src/Controller/UsersController.php]: ou alors un requirement

File: src/Manager/UsersManager.php [hunk 2]
```diff
@@ -14,8 +15,16 @@ public function __construct(private UsersRepository $usersRepo)
     /**
      * {@inheritdoc}
      */
-    public function getUsersList(Clients $client)
+    public function getUsersList($client)
```
Comment (antoinetrouve) [inline src/Manager/UsersManager.php]: idem il faut typer

File: src/Manager/UsersManager.php [hunk 3]
```diff
@@ -14,8 +15,16 @@ public function __construct(private UsersRepository $usersRepo)
...
     {
         return $this->usersRepo->findByClient($client);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUserId($client, $id)
```
Comment (antoinetrouve) [inline src/Manager/UsersManager.php]: il faut typer

File: src/Manager/UsersManagerInterface.php [hunk 4]
```diff
@@ -2,12 +2,22 @@
...
 interface UsersManagerInterface
 {
     /**
-     * Method getUsersList Contains users information.
+     * Method getUsersList. contains users information.
+     *
+     * @param $client reference to identified client ID
      */
-    public function getUsersList(Clients $client);
+    public function getUsersList($client);
```
Comment (antoinetrouve) [inline src/Manager/UsersManagerInterface.php]: il faut typer

File: src/Manager/UsersManagerInterface.php [hunk 5]
```diff
@@ -2,12 +2,22 @@
...
+
+    /**
+     * Method getUserId. Contains user information / id.
+     *
+     * @param $client Reference to identified client ID
+     * @param $id Reference to Requested Id
+     */
+    public function getUserId($client, $id);
```
Comment (antoinetrouve) [inline src/Manager/UsersManagerInterface.php]: idem

File: src/Repository/UsersRepository.php [hunk 6]
```diff
@@ -75,7 +75,7 @@ public function findOneBySomeField($value): ?Users
     }
     */
 
-    public function findByClient(Clients $client)
+    public function findByClient($client)
```
Comment (antoinetrouve) [inline src/Repository/UsersRepository.php]: idem

File: src/Repository/UsersRepository.php [hunk 7]
```diff
@@ -84,4 +84,17 @@ public function findByClient(Clients $client)
             ->getQuery()
         ;
     }
+
+    public function findOneByClient($client, $id)
```
Comment (antoinetrouve) [inline src/Repository/UsersRepository.php]: idem

**Review by antoinetrouve — APPROVED** [review]

File: src/Controller/UsersController.php (hunk 1 above)
Comment (vanmarcke) [reply src/Controller/UsersController.php]: done

File: src/Manager/UsersManager.php (hunk 2 above)
Comment (vanmarcke) [reply src/Manager/UsersManager.php]: typer Clients = App\Entity\Clients

File: src/Manager/UsersManagerInterface.php (hunk 4 above)
Comment (vanmarcke) [reply src/Manager/UsersManagerInterface.php]: typer Clients = App\Entity\Clients

File: src/Manager/UsersManager.php (hunk 3 above)
Comment (vanmarcke) [reply src/Manager/UsersManager.php]: done

File: src/Manager/UsersManagerInterface.php (hunk 5 above)
Comment (vanmarcke) [reply src/Manager/UsersManagerInterface.php]: done

File: src/Repository/UsersRepository.php (hunk 6 above)
Comment (vanmarcke) [reply src/Repository/UsersRepository.php]: done

File: src/Repository/UsersRepository.php (hunk 7 above)
Comment (vanmarcke) [reply src/Repository/UsersRepository.php]: done

**Pull request merged**